agent/http_test.go

```diff
@@ -1198,16 +1198,46 @@ func TestACLResolution(t *testing.T) {
 func TestEnableWebUI(t *testing.T) {
 	t.Parallel()
 	a := NewTestAgent(t, `
-		ui = true
+		ui_config {
+			enabled = true
+		}
 	`)
 	defer a.Shutdown()
 
 	req, _ := http.NewRequest("GET", "/ui/", nil)
 	resp := httptest.NewRecorder()
 	a.srv.handler(true).ServeHTTP(resp, req)
-	if resp.Code != 200 {
-		t.Fatalf("should handle ui")
-	}
+	require.Equal(t, http.StatusOK, resp.Code)
+
+	// Validate that it actually sent the index page we expect since an error
+	// during serving the special intercepted index.html in
+	// settingsInjectedIndexFS.Open will actually result in http.FileServer just
+	// serving a plain directory listing instead which still passes the above HTTP
+	// status assertion. This comment is part of our index.html template
+	require.Contains(t, resp.Body.String(), `<!-- CONSUL_VERSION:`)
+}
+
+func TestEnableWebUIWithMetricsOptions(t *testing.T) {
+	t.Parallel()
+	a := NewTestAgent(t, `
+		ui_config {
+			enabled = true
+			metrics_provider_options_json = "{\"foo\": 1}"
+		}
+	`)
+	defer a.Shutdown()
```

Comment:
We don't need a full running agent to test the `HTTPServer`. I think this test could be a lot less expensive.

Reply:
I agree it could for sure.

Our current testing methodology in this package with a full agent is a know area we'd like to improve but that seems out of scope for a feature PR.

The `HTTPServer` is very limited but most of the methods in it call into the agent state and similar so decoupling the server from the agent state is not trivial. Even in cases like this where we could fake it, I'm not very confident that provides a very robust test. If we had a more holistic refactor - for example make the server only have access to the (reloadable) config from the agent and specific interfaces where it needs access to RPCs and Serf, then we could make tests cheaper without making them brittle but that seems out of scope and fix all the tests not just this one new one!